eth/polar/mining.go

```diff
@@ -18,26 +18,28 @@
 // MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
 // TITLE.
 
-package rpc
+package polar
 
 import (
-	"pkg.berachain.dev/polaris/eth/rpc/api"
+	"context"
+
+	"pkg.berachain.dev/polaris/eth/core"
+	"pkg.berachain.dev/polaris/eth/core/types"
 )
 
-// GetAPIs returns a list of all available APIs.
-func GetAPIs(apiBackend PolarisBackend) []API {
-	return append(GetGethAPIs(apiBackend, nil), // todo: required chain for flashbots.
-		API{
-			Namespace: "eth",
-			Service:   api.NewEthashAPI(apiBackend),
-		},
-		API{
-			Namespace: "net",
-			Service:   api.NewNetAPI(apiBackend),
-		},
-		API{
-			Namespace: "web3",
-			Service:   api.NewWeb3API(apiBackend),
-		},
-	)
+// TODO: replace this file with a proper mining object and use message passing instead of direct calls.
```

Comment:
we need a "ABCI" like messaging protocol between a host chain and the Polaris Eth framework

Reply:
basically ye